LeaderboardBackend.Test/Controllers/Helpers.cs

```diff
@@ -0,0 +1,43 @@
+using Microsoft.AspNetCore.Mvc;
+using NUnit.Framework;
+
+namespace LeaderboardBackend.Test.Controllers
+{
+	internal static class Helpers
+	{ 
+		public static void AssertResponseNotFound<T>(ActionResult<T> response)
+		{
+			var actual = response.Result as NotFoundResult;
+			Assert.NotNull(actual);
+			Assert.AreEqual(404, actual!.StatusCode);
+		}
+
+		public static void AssertResponseBadRequest<T>(ActionResult<T> response)
+		{
+			var actual = response.Result as BadRequestResult;
+			Assert.NotNull(actual);
+			Assert.AreEqual(400, actual!.StatusCode);
+		}
+
+		public static void AssertResponseForbid<T>(ActionResult<T> response)
+		{
+			var actual = response.Result as ForbidResult;
+			Assert.NotNull(actual);
+			// ForbidResult inherits from ActionResult, not StatusCodeResult. Unlike
+			// the previous assertions, this is sufficient to know we have received
+			// a Forbidden 403 result.
+		}
+
+		public static TResult? GetValueFromObjectResult<
+			TObjectResult, 
+			TResult
+		>(ActionResult<TResult> result) where TObjectResult : ObjectResult 
+		{
+			Assert.NotNull(result);
+			var objectResult = (TObjectResult?)result.Result;
```

Comment:
If result.Result cannot be cast to whatever `TObjectResult` is, it will throw an exception. I can find a nicer way to write this that results in a nicer error message.

Reply:
Oh actually I just realized, because of the constraint I added to `TObjectResult` I can switch this to using the [as operator](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/type-testing-and-cast#as-operator) which won't throw an exception. Easier to read too. I'll do that now.

Reply:
Okay this is annoying, the [InvalidCastException](https://docs.microsoft.com/en-us/dotnet/api/system.invalidcastexception?view=net-6.0) doesn't actually provide a mechanism to extract the names of the two types, but it does expose it in the message. So what I'm gonna do is use the exception message in an assert failure. Final answer.